server/app/models/stack.rb

```diff
@@ -5,6 +5,7 @@ class Stack
   NULL_STACK = 'null'.freeze
 
   field :name, type: String
+  field :parent_name, type: String
```

Comment:
Yes that would be nice but the children exist before parents so it's a bit difficult. (well creating the parent could update the children's parent_id)

Reply:
Right, there's a fundamental choice here between letting the parent stack link to child stack services, or letting the child stack services link to the parent stack. You can't have both.

The current order of creating the child stacks before the parent stacks is probably the more natural order, because that means that something like an application service in the parent stack can link to / connect to a db service in a child stack. The limitation is that you can't have an LB service in the parent stack which is used to expose other services in child stacks, because the child stack services will not be able to link to the LB in the parent stack. You can still have an LB as a dependent stack, and have other dependent stacks link to that... or you could, if inter-dependency variables worked :confounded:

It feels a little dirty to create child stacks with these dangling `parent` links... it'll show up as `null` in the API JSON, until some suitably named stack gets created, and then it will magically change to `{"name": "..."}` :confused: The child stacks will also be left orphaned if the `kontena stack install` fails halfway-through and the parent stack doesn't get created, which means that you have to clean up all the child stacks individually with `kontena stack rm` before you can re-install the parent stack... or you install some other stack with the same name.

However, the deferred `parent_name` linking is simpler all around than setting the child `parent` relations once the parent stack gets created... maybe this isn't worth fixing until it actually becomes a problem.

Reply:
And maybe the LB usecase is more an example of the other kind of "external" dependencies, where the required LB stack would be assumed to already exist when creating the stack that requires it, and multiple installed stacks could use the same external LB stack. In that kind of inversed scenario the relations will be much more naturally modeled as mongoid relations.